common/frontend/config.js

```diff
@@ -1,4 +1,4 @@
-const CONTRACT_NAME = process.env.CONTRACT_NAME ||'near-blank-project'
+const CONTRACT_NAME = process.env.CONTRACT_NAME || process.env.VUE_APP_CONTRACT_NAME || 'near-blank-project'
```

Comment:
It's a Vue3 security issue. I tried several possible solutions and this was the most elegant. 
Vue3, as a security measure, **only imports process.env variables starting with VUE_APP_** to avoid accidentally exposing a private key. See https://cli.vuejs.org/guide/mode-and-env.html#environment-variables

Reply:
Maybe we should move `config.js` into each individual template, rather than putting it in `common/frontend`

My medium-term goal is to obsolesce this file entirely by putting sensible defaults in `near-ap-js` and using `.env` for `CONTRACT_NAME`, at which point we'll be able to get rid of the largely-duplicate `config.js` files in each template directory quite easily.

Are you up for doing this move now?

Reply:
Let's close this vue-related issue first, and then I can help you with that.

Maybe we can generate a file in /src/generated/config.js (.gitignoring src/generated) as part of the common build process after `near dev-deploy` 
The file will only export a single object with the selected configuration (based on process.env.NODE_ENV) including the contract name from `/neardev/dev-account`
This way you can delete config.js from all templates and also only the selected config data gets baked into the web app

Reply:
I don’t like the idea of React apps or vanilla JS apps being generated with a `VUE_APP_CONTRACT_NAME` variable in them, and I think we need to fix that before we merge this. I’m totally ok adding a little duplication for now, and then we can totally remove the need for this file by putting these defaults in near-api-js

Reply:
Unfortunately that won’t be enough. `config.js` will need to be moved into each template dir and deleted from `common/frontend` entirely. The `index.js` script does not copy `config.js` from the template to avoid copying symlinks, and then it copies over the stuff in `common/frontend`. So the Vue template will end up with an incorrect config.js unless we go all the way with this. 

Seems like the Angular template needed this too (https://github.com/near/bounties/issues/24)

I’ll make a PR to add this to `master` separate from this PR, and once it gets merged you can rebase/merge it in. That should make your work easier

Reply:
I created #457, once the tests pass we can get it merged and then you can rebase/merge on latest `master`

Reply:
Merged #457

Reply:
looks like this PR from my branch: add-vue-template is still good to go, no need to rebase/merge

Reply:
note that without a rebase/merge, `common/frontend/config.js` is still on your branch, so local testing with `./index.js my-vue-app --frontend vue` does not use the correct `config.js`

tests still work because they set `CONTRACT_NAME`, not `VUE_APP_CONTRACT_NAME`. Will they still pass once they use the correct `config.js`?

Reply:
I removed `common/frontend/config.js` when you asked me to create separate configs. I'll merge anyway to be safe.